File: Server/src/main/java/dev/nacho/ghub/domain/model/dto/PartidaDTO.java
package dev.nacho.ghub.domain.model.dto;

import dev.nacho.ghub.domain.model.Partida;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.format.DateTimeFormatter;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PartidaDTO {
    private String id;
    private String anfitrionNombre;
    private String juegoNombre;
    private String estado;
    private String fechaCreacion;

    public PartidaDTO toPartidaCardDto(Partida partida) {
        return new PartidaDTO(
                partida.getId(),
                partida.getAnfitrion().getNombreUsuario(),
                partida.getJuego().getNombre(),
                partida.getEstado().name(),
                partida.getFechaCreacion().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME) // Formatea la fecha
        );
    }
}

File: FrontEnd/AndroidApp/GamingHubAndroidApp/app/src/main/java/com/example/gaminghub/data/model/PartidaDTO.kt
package com.example.gaminghub.data.model

import com.example.gaminghub.domain.modelo.PartidaCard

data class PartidaDTO(
    val id: String,
    val anfitrionNombre: String,
    val juegoNombre: String,
    val estado: String,
    val fechaCreacion: String
){
    fun toPartidaTarjeta(): PartidaCard {
        return PartidaCard(
            id = this.id,
            juegoNombre = this.juegoNombre,
            estado = this.estado,
            fechaCreacion = this.fechaCreacion)
    }
}